### PR TITLE
#521 - Auth fix for Docker repositories

### LINKED ISSUE #521: Docker repositories do not work with authentication
When Docker client tries to perform an operation it first sends `GET /v2/` request. Response to this request should contain version header and `WWW-Authenticate` header if needed. Now Artipie always sends empty `200 OK` response for this endpoint. This behavior is implemented in `DockerRoutingSlice`. It works well in scenarios when authentication is not required. But when auth is required for other endpoints client does not receive `WWW-Authenticate` and does not tries to authenticate for any other requests. So in practice authenticated access does not work for Docker repositories.

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #521 
Proper response from `/v2/` endpoint

### CODE REVIEW

File: src/test/java/com/artipie/http/DockerRoutingSliceTest.java [hunk 1]
```diff
@@ -54,19 +70,37 @@ void ignoresNonDockerRequests() throws Exception {
...
+            new SettingsWithAuth(new Authentication.Single(username, password)),
+            (line, headers, body) -> {
+                throw new UnsupportedOperationException();
+            }
+        ).response(
+            new RequestLine(RqMethod.GET, "/v2/").toString(),
+            new Headers.From(new Authorization.Basic(username, password)),
+            Content.EMPTY
+        );
+        MatcherAssert.assertThat(
```
Comment (g4s8) [inline src/test/java/com/artipie/http/DockerRoutingSliceTest.java:94]: @olegmoz you can use `SliceHasResponse` here:
```java
MatcherAssert.assertThat(
  new DockerRoutingSlice(...),
  new SliceHasResponse(
    new RsHasStatus(status),
    new RequestLine(...)
  )
)
```

**Review by g4s8 — APPROVED** [review]
@olegmoz please check my comment

**Comment by olegmoz** [comment]
@g4s8 thanks, please see the changes

**Review by g4s8 — COMMENTED** [review]
@olegmoz thanks

**Comment by olegmoz** [comment]
@g4s8 could you please approve the PR?

**Review by g4s8 — APPROVED** [review]

**Pull request merged**